spacy/tests/serialize/test_serialize_doc.py

```diff
@@ -1,13 +1,204 @@
+
+
+@pytest.mark.issue(1883)
+def test_issue1883():
+    matcher = Matcher(Vocab())
+    matcher.add("pat1", [[{"orth": "hello"}]])
+    doc = Doc(matcher.vocab, words=["hello"])
+    assert len(matcher(doc)) == 1
+    new_matcher = copy.deepcopy(matcher)
+    new_doc = Doc(new_matcher.vocab, words=["hello"])
+    assert len(new_matcher(new_doc)) == 1
+
+
+@pytest.mark.issue(2564)
+def test_issue2564():
+    """Test the tagger sets has_annotation("TAG") correctly when used via Language.pipe."""
+    nlp = Language()
+    tagger = nlp.add_pipe("tagger")
+    tagger.add_label("A")
+    nlp.initialize()
+    doc = nlp("hello world")
+    assert doc.has_annotation("TAG")
+    docs = nlp.pipe(["hello", "world"])
+    piped_doc = next(docs)
+    assert piped_doc.has_annotation("TAG")
+
+
+@pytest.mark.issue(3248)
+def test_issue3248_2():
+    """Test that the PhraseMatcher can be pickled correctly."""
+    nlp = English()
+    matcher = PhraseMatcher(nlp.vocab)
+    matcher.add("TEST1", [nlp("a"), nlp("b"), nlp("c")])
+    matcher.add("TEST2", [nlp("d")])
+    data = pickle.dumps(matcher)
+    new_matcher = pickle.loads(data)
+    assert len(new_matcher) == len(matcher)
+
+
+@pytest.mark.issue(3289)
+def test_issue3289():
+    """Test that Language.to_bytes handles serializing a pipeline component
+    with an uninitialized model."""
+    nlp = English()
+    nlp.add_pipe("textcat")
+    bytes_data = nlp.to_bytes()
+    new_nlp = English()
+    new_nlp.add_pipe("textcat")
+    new_nlp.from_bytes(bytes_data)
+
+
+@pytest.mark.issue(3468)
+def test_issue3468():
+    """Test that sentence boundaries are set correctly so Doc.has_annotation("SENT_START") can
+    be restored after serialization."""
+    nlp = English()
+    nlp.add_pipe("sentencizer")
+    doc = nlp("Hello world")
+    assert doc[0].is_sent_start
+    assert doc.has_annotation("SENT_START")
+    assert len(list(doc.sents)) == 1
+    doc_bytes = doc.to_bytes()
+    new_doc = Doc(nlp.vocab).from_bytes(doc_bytes)
+    assert new_doc[0].is_sent_start
+    assert new_doc.has_annotation("SENT_START")
+    assert len(list(new_doc.sents)) == 1
+
+
+@pytest.mark.issue(3959)
+def test_issue3959():
+    """Ensure that a modified pos attribute is serialized correctly."""
+    nlp = English()
+    doc = nlp(
+        "displaCy uses JavaScript, SVG and CSS to show you how computers understand language"
+    )
+    assert doc[0].pos_ == ""
+    doc[0].pos_ = "NOUN"
+    assert doc[0].pos_ == "NOUN"
+    # usually this is already True when starting from proper models instead of blank English
+    with make_tempdir() as tmp_dir:
+        file_path = tmp_dir / "my_doc"
+        doc.to_disk(file_path)
+        doc2 = nlp("")
+        doc2.from_disk(file_path)
+        assert doc2[0].pos_ == "NOUN"
+
+
+@pytest.mark.issue(4367)
+def test_issue4367():
+    """Test that docbin init goes well"""
+    DocBin()
+    DocBin(attrs=["LEMMA"])
+    DocBin(attrs=["LEMMA", "ENT_IOB", "ENT_TYPE"])
```

Comment:
It looks like we've been putting `DocBin` tests into this file because it's a kind of serialization of `Doc`'s, but this doesn't feel quite right. Perhaps we should put all the docbin stuff into its own module? Maybe we could even have `tests/doc/test_docbin` and `tests/serialize/test_serialize_docbin.py`

Reply:
Agree, will do that :)

Reply:
Changeset here: https://github.com/explosion/spaCy/pull/9655/commits/e68ee5cae53faf01b6009e1649287235c570bde1

I created a new file: `test_serialize_docbin.py` as a container for all DocBin related to tests. I also went away and moved non-regression `DocBin` tests from `test_serialize_doc.py` to this new file.